src/rust/engine/client/src/options/args.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+use std::env;
+
+use super::id::{NameTransform, OptionId, Scope};
+use super::parse::parse_bool;
+use super::OptionsSource;
+
+pub(crate) struct Args {
+  args: Vec<String>,
+}
+
+enum Negate {
+  TRUE,
+  FALSE,
+}
+
+impl Args {
+  pub(crate) fn argv() -> Args {
+    Args {
+      args: env::args().collect::<Vec<_>>(),
+    }
+  }
+
+  fn arg_name(id: &OptionId, negate: Negate) -> String {
+    format!(
+      "--{}{}{}",
+      match negate {
+        Negate::FALSE => "",
+        Negate::TRUE => "no-",
+      },
+      match &id.0 {
+        Scope::GLOBAL => "".to_string(),
+        Scope::Scope(scope) => format!("{}-", scope.to_ascii_lowercase()),
+      },
+      id.name("-", NameTransform::ToLower)
+    )
+  }
+
+  fn find_string(&self, arg_name: &str) -> Result<Option<String>, String> {
+    for arg in &self.args {
+      let mut components = arg.as_str().splitn(2, '=');
+      if let Some(name) = components.next() {
+        if name == arg_name {
+          return Ok(Some(components.next().unwrap_or("").to_owned()));
+        }
+      }
+    }
+    Ok(None)
+  }
+}
```

Comment:
This will find the first instance of an argument, but you probably need to find the last instance: i.e., for:
```
./pants --pantsd --no-pantsd
./pants --pants-distdir=x --pants-distdir=y
```
... the latter should win.

Reply:
Yup, fixed. Still needs tests though.